loadbalancer/algorithm.go

```diff
@@ -190,31 +191,51 @@ func (r *random) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 	return withFadeIn(r.rand, ctx, r.notFadingIndexes, r.fadingWeights, i)
 }
 
-type consistentHash struct{}
+type (
+	endpointHash struct {
+		index int    // index of endpoint in endpoint list
+		hash  uint32 // hash of endpoint
+	}
+	consistentHash []endpointHash // list of endpoints sorted by hash value
+)
+
+func (ch consistentHash) Len() int           { return len(ch) }
+func (ch consistentHash) Less(i, j int) bool { return ch[i].hash < ch[j].hash }
+func (ch consistentHash) Swap(i, j int)      { ch[i], ch[j] = ch[j], ch[i] }
 
 func newConsistentHash(endpoints []string) routing.LBAlgorithm {
-	return &consistentHash{}
+	ch := consistentHash(make([]endpointHash, len(endpoints)))
+	for i, ep := range endpoints {
+		ch[i] = endpointHash{i, hash(ep)}
+	}
+	sort.Sort(ch)
```

Comment:
This might benefit from `sort.Stable()`, wdyt?

Reply:
The actual order does not really matter as long key search leads to the same endpoint consistently. Also it is sorted by hash value which (pseudo-)random.

Reply:
But I see a an edge case here: when two endpoints happen to have the same hash, the first will receive requests and the second will not.

Reply:
> But I see a an edge case here
 
Strictly speaking consistent hash *does not guarantee* neither even request distribution across endpoints nor that every endpoint receives requests at all. 

E.g. backend with two endpoints may end up with different hashes but a very narrow hash difference (e.g. 1) by a bad luck and most/all of the requests would go to the first (with lower hash value) endpoint as the request key hash would be either less than the hash of the first endpoint or greater than the hash of the second endpoint (due to low chance to happen right inbetween) and thus rollover to the first.

Reply:
ok, seems fine